tribe-ext-dequeue-assets/index.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * Plugin Name: Tribe Extension: Dequeque Assets
+ * Description: Dequeues any scripts or styles that are registered using the tribe_asset() function. Adds a list of dequeueable assets to WP Admin > Events > Settings.
+ * Version: 1.0.0
+ * Extension Class: Tribe__Extension__Dequeue_Assets
+ * Author: Modern Tribe, Inc.
+ * Author URI: http://m.tri.be/1971
+ * License: GPLv2 or later
+ * License URI: https://www.gnu.org/licenses/gpl-2.0.html
+ */
+
+// Do not load directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+// Do not load unless Tribe Common is fully loaded.
+if ( ! class_exists( 'Tribe__Extension' ) ) {
+	return;
+}
+
+/**
+ * Extension main class, class begins loading on init() function.
+ */
+class Tribe__Extension__Dequeue_Assets extends Tribe__Extension {
+
+	/**
+	 * All assets that have been detected on this install
+	 *
+	 * @var array
+	 */
+	protected $logged_assets = array();
+
+	/**
+	 * Any asset meant to be dequeued
+	 *
+	 * @var array
+	 */
+	protected $dequeued_assets = array();
+
+	/**
+	 * Holds the instance of Tribe__Assets
+	 *
+	 * @var Tribe__Assets Instance
+	 */
+	protected $ta_instance;
+
+	/**
+	 * Option key containing all logged assets on install
+	 *
+	 * @var string
+	 */
+	protected $logged_assets_key = 'logged_assets';
+
+	/**
+	 * Option key for dequeued assets
+	 *
+	 * @var string
+	 */
+	protected $dequeued_assets_key = 'dequeued_assets';
+
+	/**
+	 * Prefix used for reset URLs
+	 *
+	 * @var string Prefix
+	 */
+	protected $reset_url_parameter_prefix = 'tribe_extension_reset_';
+
+	/**
+	 * Setup the Extension's properties.
+	 *
+	 * This always executes even if the required plugins are not present.
+	 */
+	public function construct() {
+		$this->set_url( 'https://theeventscalendar.com/extensions/dequeue-assets/' );
+		// Set this to init() straight away, so we can begin logging assets.
+		$this->set( 'init_hook', 'plugins_loaded' );
```

Comment:
We need to require a bigger version, where `Tribe__Utils__Array` was introduced.